File: exercises/Ex04StringManipulation/string-manipulation.js

const app = require("express")()
const bodyParser = require("body-parser")

app.use(bodyParser.raw({ type: "*/*" }))

app.post("/string-manipulation", (req, res) => {
  let result
  const body = req.body.toString()
  const search = "I'm positive"

  if (body.slice(0, search.length) === search) {
    result = "I think" + body.substr(search.length)
  }
  else {
    result = body
  }

  res.send(result)
})

console.log("Starting server...")
app.listen(7879)

File: exercises/Ex04StringManipulation/StringManipulation.hs
module Ex04StringManipulation.StringManipulation where

import qualified Data.List as List
import qualified Zero.Server as Server


stringManipulationHandler :: Server.Request -> Server.Response
stringManipulationHandler req
  = Server.stringResponse res

  where
    search
      = "I'm positive"

    len
      = List.length search

    body
      = Server.requestBody req

    res
      = if List.take len body == search
          then "I think" <> List.drop len body
          else body

main :: IO ()
main
  = Server.startServer
      [ Server.simpleHandler Server.POST "/string-manipulation" stringManipulationHandler
      ]
